### PR TITLE
Feat/poll modifications

### PR DESCRIPTION
**Pull request by timur-ul-hasan**
Changes:
- Add approval_minimum field into model and added the flow for this in create poll API.
- Add finalization_period into model and added the flow for this in create poll API.

### CODE REVIEW

File: flowback/poll/models.py [hunk 1]
```diff
@@ -86,6 +86,21 @@ class PollType(models.IntegerChoices):
     participants = models.IntegerField(default=0)
     dynamic = models.BooleanField()
 
+    approval_minimum = models.IntegerField(
+        validators=[MinValueValidator(0), MaxValueValidator(100)],
```
Comment (Gofven) [inline flowback/poll/models.py:104]: MaxValueValidator at 100% guarantees no votes will pass, change this to 99 c:

**Review by Gofven — CHANGES_REQUESTED** [review]
Aside from one needed change, well done! I'll merge this branch immediately once you've made that tiny change c:

edit: Checked with the project manager, 100% is all good!

**Comment by timur-ul-hasan** [comment]
Hi @Gofven 
I have updated the pull request with the flow for dynamically updating the poll in the celery task, which will run daily.

File: flowback/poll/services/vote.py [hunk 2]
```diff
@@ -335,3 +361,35 @@ def save_participants_count(voting_type_class):
...
+
+    if poll.dynamic:
+        quorum_completed = poll.participants > total_group_users * quorum
+
+        if quorum_completed:
+            if poll.approval_minimum:
+                approval_minimum = poll.approval_minimum / 100
+                # Check if the approval_minimum is met
+                approval_minimum_completed = False
+                for proposal in poll.pollproposal_set.all():
```
Comment (Gofven) [inline flowback/poll/services/vote.py]: Would be better to use aggregate here c:

File: flowback/poll/services/vote.py [hunk 3]
```diff
@@ -335,3 +361,35 @@ def save_participants_count(voting_type_class):
...
+                    proposal_positive_votes = proposal.proposal_votes_info.get('positive_votes', 0)
+                    if proposal_positive_votes > poll.participants * approval_minimum:
```
Comment (Gofven) [inline flowback/poll/services/vote.py]: poll.participants * approval_minimum isn't correct, you should multiply approval_minimum with the total amount of voters

**Review by Gofven — CHANGES_REQUESTED** [review]

File: flowback/poll/services/vote.py (hunk 3 above)
Comment (the-hasan-javed) [reply flowback/poll/services/vote.py]: @Gofven 
**poll. participants** are the total amount of votes, they are calculated before this point using this query:

```
   poll.participants = (
                mandate + PollVoting.objects.filter(poll=poll).all().count()
            )
```

File: flowback/poll/services/vote.py (hunk 2 above)
Comment (the-hasan-javed) [reply flowback/poll/services/vote.py]: Sure @Gofven

**Pull request closed without merging**